pkg/kapp/app/recorded_app.go

```diff
@@ -146,6 +150,7 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string, isDiffRun bool) (
 	}
 
 	if foundNonMigratedApp {
+		_, a.isAppLabelAppChangeLabel = app.Annotations[KappIsAppLabelAsAppChangeLabelKey]
```

Comment:
Yeah, I was thinking of keeping it outside as well, the only reason I didn't do it is because if `!foundNonMigratedApp` then app would also be empty, but that should be okay, so I am totally fine with making this change.

Reply:
Maybe do it after a check on `app` ? That would make sense I think

Reply:
The function `RenamePrevApp` seems to have similar logic and should also reflect this change?

Reply:
> Maybe do it after a check on app ?

We return from the check on app.